.circleci/config.yml

```diff
@@ -584,6 +620,34 @@ jobs:
             docker push ${DOCKERHUB_REPO}:nimbus-firefox-beta
             docker push ${DOCKERHUB_REPO}:nimbus-firefox-release
 
+  build_firefox_fenix:
+    working_directory: ~/experimenter
+    machine:
+      image: ubuntu-2004:2023.10.1
+      docker_layer_caching: true
+    resource_class: xlarge
+    steps:
+      - checkout
+      - docker_login:
+          username: $DOCKER_USER
+          password: $DOCKER_PASS
+      - run:
+          name: Build and upload images
+          command: |
+            cd experimenter/tests/integration/nimbus/android
+            docker run --name fenix-builder ${DOCKERHUB_REPO}:experimenter-moz-central bash build_android.sh
+            docker cp fenix-builder:mozilla-central/mobile/android/fenix/app-fenix-debug-androidTest.apk ./
+            docker cp fenix-builder:mozilla-central/mobile/android/fenix/app-fenix-x86_64-debug.apk ./
+            docker commit fenix-builder moz-central-builder
+            docker tag moz-central-builder ${DOCKERHUB_REPO}:experimenter-moz-central
+            docker push ${DOCKERHUB_REPO}:experimenter-moz-central
+          no_output_timeout: 1h
+      - save_cache:
+          key: experimenter-firefox-fenix-apks-{{ checksum "experimenter/tests/integration/nimbus/android/app-fenix-x86_64-debug.apk" }}
+          paths:
+            - experimenter/tests/integration/nimbus/android/app-fenix-debug-androidTest.apk
+            - experimenter/tests/integration/nimbus/android/app-fenix-x86_64-debug.apk
```

Comment:
Push to dockerhub is just mozzila-central with a `./mach build` completed, so that next time we just do a `hg pull` and `hg update` and we don't have to clone the entire repo. Caching the APKs is for the actual test job.